components/developer/jenkins-swarm-client/files/jenkins-swarm-client.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+
+# Start the swarm client, optionally with configs prepared by some helper
+#
+# This file and its contents are supplied under the terms of the Jenkins
+# license (a derivative of MIT license).
+# Copyright 2016-2021 Jim Klimov
+#
+
+#trap '_FINAL=$? ; echo "Exiting with code ${_FINAL}" >&2; sleep 5; exit ${_FINAL}' 0 1 2 3 15
+#echo "`date -u` : starting $0 $*" >&2 ; set
+#set -x
+set -o pipefail
+
+TZ=UTC
+LANG=C
+LC_ALL=C
+export TZ LANG LC_ALL
+
+# Failsafe against accidental runs
+if [ $# = 0 ] ; then
+    echo "ERROR: An argument is required" >&2
+    exit 1
+fi
+case "$1" in
+    start) ;;
+    #stop|refresh|restart) ;& # these are handled by SMF
+    *) echo "ERROR: Argument is currently not supported: $1" >&2 ; exit 1 ;;
+esac
+shift
+
+# Default setting: Allow to scale RAM consumption
+JAVA_OPTS="-Xms64m -Xmx512m"
+
+# Allow to investigate why an agent crashed, which rarely happens, to report
+# details into  https://issues.jenkins-ci.org/browse/JENKINS-60673
+JAVA_OPTS="$JAVA_OPTS -XX:+HeapDumpOnOutOfMemoryError"
+
+# This script can be used standalone, or as an SMF method script
```

Comment:
I think a "dual mode" like this is not such a good idea. If we are using SMF it should call a simple wrapper in /usr/bin to handle application specific startup and handle the SMF part in the method file. Keep things where they belong.

Reply:
Repeating the misplaced comment form another conversation: this script originated from, and hopefully will remain usable as, a script for setting up the Jenkins Swarm Client on a large variety of OSes. So I am not in favor of tailoring it to any single OS and/or service management framework (that is, in a way that reduces its usability "as is" in other cases).